src/server/game/Scripting/ScriptMgr.cpp

```diff
@@ -2179,220 +2188,1098 @@ void ScriptMgr::OnBeforeMailDraftSendMailTo(MailDraft* mailDraft, MailReceiver c
+}
+
+AuctionHouseScript::AuctionHouseScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<AuctionHouseScript>::AddScript(this);
+}
+
+ConditionScript::ConditionScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<ConditionScript>::AddScript(this);
+}
+
+VehicleScript::VehicleScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<VehicleScript>::AddScript(this);
+}
+
+DynamicObjectScript::DynamicObjectScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<DynamicObjectScript>::AddScript(this);
+}
+
+TransportScript::TransportScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<TransportScript>::AddScript(this);
+}
+
+AchievementCriteriaScript::AchievementCriteriaScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<AchievementCriteriaScript>::AddScript(this);
+}
+
+PlayerScript::PlayerScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<PlayerScript>::AddScript(this);
+}
+
+AccountScript::AccountScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<AccountScript>::AddScript(this);
+}
+
+GuildScript::GuildScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<GuildScript>::AddScript(this);
+}
+
+GroupScript::GroupScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<GroupScript>::AddScript(this);
+}
+
+GlobalScript::GlobalScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<GlobalScript>::AddScript(this);
+}
+
+BGScript::BGScript(char const* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<BGScript>::AddScript(this);
+}
+
+ArenaTeamScript::ArenaTeamScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<ArenaTeamScript>::AddScript(this);
+}
+
+SpellSC::SpellSC(char const* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<SpellSC>::AddScript(this);
+}
+
+ModuleScript::ModuleScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<ModuleScript>::AddScript(this);
+}
+
+GameEventScript::GameEventScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<GameEventScript>::AddScript(this);
+}
+
+MailScript::MailScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<MailScript>::AddScript(this);
+}
+
+AchievementScript::AchievementScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<AchievementScript>::AddScript(this);
+}
+
+PetScript::PetScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<PetScript>::AddScript(this);
+}
+
+ArenaScript::ArenaScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<ArenaScript>::AddScript(this);
+}
+
+MiscScript::MiscScript(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<MiscScript>::AddScript(this);
+}
+
+CommandSC::CommandSC(const char* name)
+    : ScriptObject(name)
+{
+    ScriptRegistry<CommandSC>::AddScript(this);
```

Comment:
this file is getting HUGE, wouldn't be better to split it?

Reply:
Better to open another PR/issue for it. We cannot increase the complexity of this one now